### PR TITLE
plant_object.rb Submitting plant_object.rb for approval.

### PR DESCRIPTION
**Pull request by danadaldos**
[#89155666]

### CODE REVIEW

File: objects/plant_object.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env ruby
+# This class mimics a houseplant
```
Comment (jasonnoble) [inline objects/plant_object.rb]: Usually a blank line after the shebang line.

File: objects/plant_object.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+# You can <.fertilize> the plant
+# You can <.wait> one week to see what happens.
+# If the soil is wetter than a 2 and you have fertilized the plant, it will
+# grow a new leaf.
+# You can use <.count_the_leaves> to see how many leaves the plant has.
+class Plant
+  def initialize
+    moisture = 0
+    @leaves = 3
+    @moisture = @moisture.to_i + moisture
```
Comment (jasonnoble) [inline objects/plant_object.rb]: Why not just `@moisture = 0`?

Comment (jasonnoble) [inline objects/plant_object.rb]: If you make this change, you can get rid of line 14.

File: objects/plant_object.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  end
+
+  def count_the_leaves
+    puts "Plant has #{@leaves} leaves on it."
+  end
+
+  def color_check
+    if @fertilizer == 1
+      puts 'The plant looks green and healthy.'
+    elsif @fertilizer != 1
```
Comment (jasonnoble) [inline objects/plant_object.rb]: This should be an `else`.  There's only two options here, `@fertilizer` is one, or it's not.

File: objects/plant_object.rb [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    puts 'You whisper nice things to the plant. The plant is very happy'
+  end
+
+  def wait
+    puts "You wait one week."
+    if @moisture >= 2 && @fertilizer >= 1
+      puts "Plant has grown one new leaf"
+      @leaves += 1
+    end
+    @moisture -= @moisture
```
Comment (jasonnoble) [inline objects/plant_object.rb]: Why not just set them to 0?

File: objects/plant_object.rb (hunk 1 above)
Comment (danadaldos) [inline objects/plant_object.rb]: Done.

File: objects/plant_object.rb (hunk 2 above)
Comment (danadaldos) [inline objects/plant_object.rb]: Done.

File: objects/plant_object.rb (hunk 3 above)
Comment (danadaldos) [inline objects/plant_object.rb]: I want this to handle the situation of overfertilizing. Fertilizer should be exactly one, not over and not under. Between ".wait" periods, someone could .fertilize twice or more and that should also result in the "The plant looks yellow and unhealthy" message.

File: objects/plant_object.rb (hunk 4 above)
Comment (danadaldos) [inline objects/plant_object.rb]: I don't understand. The conditions for a new leaf need to be @moisture >= 2 and @fertilizer >=1.

**Comment by jasonnoble** [comment]
@wb9753 @danadaldos delete the branch this PR was based on (misnamed branch).  That's why this PR shows as closed.

**Pull request closed without merging**